tests/frameworks/test_aioamqp.py

```diff
@@ -0,0 +1,128 @@
+import asyncio
+from typing import Any, Generator
+
+import aioamqp
+import pytest
+
+from instana.singletons import tracer
+from tests.helpers import testenv
+from aioamqp.properties import Properties
+from aioamqp.envelope import Envelope
+
+testenv["rabbitmq_host"] = "127.0.0.1"
+testenv["rabbitmq_port"] = 5672
+
+
+class TestAioamqp:
+    @pytest.fixture(autouse=True)
+    def _resource(self) -> Generator[None, None, None]:
+        self.recorder = tracer.span_processor
+        self.recorder.clear_spans()
+
+        self.loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(None)
+        yield
+        self.loop.run_until_complete(self.delete_queue())
+        if self.loop.is_running():
+            self.loop.close()
+
+    async def delete_queue(self) -> None:
+        transport, protocol = await aioamqp.connect(
+            testenv["rabbitmq_host"],
+            testenv["rabbitmq_port"],
+        )
+        channel = await protocol.channel()
+        await channel.queue_delete("message_queue")
+        await asyncio.sleep(1)
+
+    async def publish_message(self) -> None:
+        transport, protocol = await aioamqp.connect(
+            testenv["rabbitmq_host"],
+            testenv["rabbitmq_port"],
+        )
+        channel = await protocol.channel()
+
+        await channel.queue_declare(queue_name="message_queue")
+
+        message = "Instana test message"
+        await channel.basic_publish(
+            message.encode(), exchange_name="", routing_key="message_queue"
+        )
+
+        await protocol.close()
+        transport.close()
+
+    async def consume_message(self) -> None:
+        async def callback(
+            channel: Any,
+            body: bytes,
+            envelope: Envelope,
+            properties: Properties,
+        ) -> None:
+            with tracer.start_as_current_span("callback-span"):
+                await channel.basic_client_ack(delivery_tag=envelope.delivery_tag)
+
+        _, protocol = await aioamqp.connect(
+            testenv["rabbitmq_host"], testenv["rabbitmq_port"]
+        )
+        channel = await protocol.channel()
+        await channel.queue_declare(queue_name="message_queue")
+        await channel.basic_consume(callback, queue_name="message_queue", no_ack=False)
+
+    def test_basic_publish(self) -> None:
+        with tracer.start_as_current_span("test-span"):
+            self.loop.run_until_complete(self.publish_message())
+
+        spans = self.recorder.queued_spans()
+
+        assert len(spans) == 2
+        aioamqp_span = spans[0]
+        test_span = spans[1]
+
+        assert aioamqp_span.n == "aioamqp-publisher"
```

Comment:
Is `aioamqp-publisher` (and also `aioamqp-consumer`) a valid entry that the Instana backend understands? The backend uses the Span Name (`span.n`) to process the trace correlation, and if the name is not supported, it will raise an error on the backend side.

Reply:
Checked backend and removed from entry-exit spans list